src/App.jsx

```diff
@@ -14,33 +14,82 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
+const prepareGoods = (goods, sortType, isReversed) => {
+  const sortedGoods = [...goods];
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+  if (sortType === 'alphabet') {
+    sortedGoods.sort((a, b) => a.localeCompare(b));
+  } else if (sortType === 'length') {
+    sortedGoods.sort((a, b) => a.length - b.length);
+  }
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+  if (isReversed) {
+    sortedGoods.reverse();
+  }
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
+  return sortedGoods;
+};
+
+export const App = () => {
+  const [sortType, setSortType] = useState('');
+  const [isReversed, setIsReversed] = useState(false);
+
+  const preparedGoods = prepareGoods(goodsFromServer, sortType, isReversed);
+
+  const isGoodsChanged = sortType !== '' || isReversed;
+
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          data-cy="SortAlphabeticallyButton"
+          type="button"
+          className={`button is-info ${sortType === 'alphabet' ? '' : 'is-light'}`}
+          onClick={() => setSortType('alphabet')}
```

Comment:
Consider adding a condition to toggle the sort type off if the same button is clicked again. This will prevent the sort type from being set repeatedly to the same value, which might be unnecessary and could lead to unexpected behavior if additional logic is added later.